### PR TITLE
Modified hotfix for something funny with sockets/sessions, updated licen...

### PR DESCRIPTION
**Pull request by sbromberger**
...se

this works

cleanup

possible readall() fix

irb/tempfix plus some other functionality

see, e.g., https://github.com/osstech-jp/openldap/commit/6ad808dc1db41fabf29a80842649f1105782f890

typo in license

removed commented code

### CODE REVIEW

**Comment by IainNZ** [comment]
Looks like we also need #35 in there to get it passing on 0.4

**Comment by sbromberger** [comment]
did you confirm that? I was unable to reproduce #35 in 0.4.

**Comment by IainNZ** [comment]
See: https://travis-ci.org/JuliaWeb/GnuTLS.jl/jobs/51184449

**Comment by sbromberger** [comment]
That's very weird. Could it be because the version of libgnutls is so old?

File: src/GnuTLS.jl [hunk 1]
```diff
@@ -8,16 +8,33 @@ include("../deps/deps.jl")
...
 
-
+const GNUTLS_MINSECURE_VER = @compat Dict(
+	v"2.12"	=>	23,
+	v"3.1"	=>	28,
+	v"3.2"	=>	20,
+	v"3.3"	=>	10
+)
+
+const WRITE_RETURN_TYPE = VERSION >= v"0.4-" ? UInt64 : Int64
```
Comment (staticfloat) [inline src/GnuTLS.jl]: I was scared to do this since `UInt64` is nothing on 0.3.  We'll see what happens, I suppose!  :D

Comment (sbromberger) [inline src/GnuTLS.jl]:
```
julia> using Compat

julia> typeof(uint64(1))
Uint64

julia> typeof(int64(1))
Int64

julia> typeof(UInt64)
DataType

julia> UInt64 == Uint64
true

julia> Base.versioninfo()
Julia Version 0.3.5
Commit a05f87b* (2015-01-08 22:33 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin14.0.0)
  CPU: Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

What am I missing?

Comment (staticfloat) [inline src/GnuTLS.jl]: Nothing!  `Compat` fixes things perfectly.

**Comment by sbromberger** [comment]
As an aside, why are there two (identical?) travis jobs being submitted?

**Comment by staticfloat** [comment]
I believe one builds this PR as-is, and one builds this PR after merging it into `master`.

**Comment by sbromberger** [comment]
All checks passed. I'll go ahead and merge.

**Comment by staticfloat** [comment]
Just for posterity's sake, I did a bisect session and discovered that the `Base.write()` stuff was because of https://github.com/JuliaLang/julia/commit/2af73dd7723c91b0e9fcf5af7662b40c8fb97067

**Comment by sbromberger** [comment]
@staticfloat thanks for checking. When was that change made? I see it was authored 29 days ago, but on my 14-day-old build I didn't have a problem.

**Pull request merged**